src/start.ts

```diff
@@ -50,18 +50,20 @@ export const handler = async (_event: any = {}): Promise<any> => {
     }),
     triggers: { ...nodeConfig.triggers, ...keeperTriggers },
   };
-  const { chains, nodeSettings, triggers, ois: oises, apiCredentials } = config;
+  const { chains, triggers, ois: oises, apiCredentials } = config;
+
+  const airnodeHDNode = ethers.utils.HDNode.fromExtendedKey(airnodeXpub);
+  if (airnodeAddress !== airnodeHDNode.derivePath('0/0').address) {
+    throw new Error(`xpub does not belong to Airnode: ${airnodeAddress}`);
+  }
```

Comment:
Nice! I see this is the same code form airnode-admin-cli [verifyAirnodeAddress()](https://github.com/api3dao/airnode/blob/v1-protocol/packages/airnode-admin/src/implementation.ts#L41) function 👌🏻